lib/curl_addrinfo.c

```diff
@@ -50,6 +50,10 @@
 #  define in_addr_t unsigned long
 #endif
 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
```

Comment:
Please also check if compiling with `USE_UNIX_SOCKETS` otherwise it won't compile on Windows systems that don't have the header

Reply:
I made an experiment and targeted the very first Windows 10 version (`10.0.10240.0`) and tried to compile curl:
![image](https://user-images.githubusercontent.com/14150058/117059489-86651280-ad28-11eb-80cb-7d8d6d771667.png)

Without a fix I am getting the following failure:
```c
curl_addrinfo.c(483,8): error C2027: use of undefined type 'sockaddr_un'
```

With a fix applied source compiles successfully. So targeting the initial Windows 10 version (`10.0.10240.0`) with the standard latest `Windows 10 SDK (10.0.19041.0)` shipped with Visual Studio works fine.

Reply:
Added the proposed additional check for `USE_UNIX_SOCKETS`.

Reply:
Please run checksrc
~~~
./curl_addrinfo.c:53:96: warning: Longer than 79 columns (LONGLINE)
 #if defined(USE_UNIX_SOCKETS) && defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
~~~

Reply:
Okay as long as the CPP macros `WINAPI_FAMILY` and `WINAPI_FAMILY_APP` you have my go 👍

Reply:
`(WINAPI_FAMILY == WINAPI_FAMILY_APP)` is valid only for Windows 10 UWP platform and both defines are declared in `winapifamily.h`:

It is used for quite long time in PortAudio project without any issue for several years:
https://github.com/PortAudio/portaudio/blob/39bfe6c63df5b85bb651f16327e7386504089276/src/hostapi/wasapi/pa_win_wasapi.c#L68